### PR TITLE
Version 1.5.0

### PR DESCRIPTION
**Pull request by manoskary**
# PR for version 1.5.0

This new version addresses multiple changes, bug fixes and new features:

### New features

- DCML format input in #345 
- New kern import for faster and more robust in #344
- Barebones Kern export in #344
- MEI export in #347 and #351 
- Estimate symbolic durations update in #352 
- New harmony classes and checks for Roman numerals, Chord Symbols, Cadences and Phrases in #345 
- Intervals as partitura classes in #345 
- transposition of parts in #345 
- Export wav with fluidsynth #353 

### Other Changes
- improved documentation
- improved typing
- New tests
- optional dependency of pandas

###

### CODE REVIEW

File: partitura/io/exportkern.py [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+        self.out_data[0] = "**kern"
+        self.out_data[-1] = "*-"
+        # Add the staff element to the second line
+        for i in range(self.unique_voc_staff.shape[0]):
+            self.out_data[1, i] = f"*staff{self.unique_voc_staff[i][1]}"
+        self.prev_note_time = None
+        self.prev_note_col_idx = None
+        self.prev_note_row_idx = None
+
+    def parse(self):
```
Comment (fosfrancesco) [inline partitura/io/exportkern.py:106]: what is a `parse` function doing in an export? Some doc for this function will help a lot!

File: partitura/utils/music.py [hunk 2]
```diff
@@ -471,6 +193,88 @@ def _transpose_note(note, interval):
         )
 
 
+def transpose_note_old(step, alter, interval):
```
Comment (fosfrancesco) [inline partitura/utils/music.py:196]: is this function still useful? Or can it be deleted?

File: partitura/utils/music.py [hunk 3]
```diff
@@ -471,6 +193,88 @@ def _transpose_note(note, interval):
...
+            diff_sm = prev_pc - tmp_pc if prev_pc >= tmp_pc else prev_pc + 12 - tmp_pc
+        new_alter = (
+            INTERVAL_TO_SEMITONES[interval.quality + str(interval.number)] - diff_sm
+        )
+    return new_step, new_alter
+
+
+def transpose_note(step, alter, interval):
+    """
+    Transpose a note by a given interval without changing the octave or creating a Note Object.
```
Comment (fosfrancesco) [inline partitura/utils/music.py:238]: I would write "without considering the octave".
And the second part "without creating a Note object is a bit weird. Maybe write in a new line "This function don't create a Note object, but just returns step and alterations attributes"

File: partitura/utils/music.py [hunk 4]
```diff
@@ -3392,6 +3279,27 @@ def tokenize(
     return tokens
 
 
+def step2pc(step, alter):
+    """
+    Convert a step to a pitch class.
```
Comment (fosfrancesco) [inline partitura/utils/music.py:3284]: New proposal for Documentation "Convert a tonal pitch class (i.e., step + alter) to a pitch class (i.e., integer in [0,11])

File: tests/test_dcml_import.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+
+class ImportDCMLAnnotations(unittest.TestCase):
+    def test_tsv_import_from_dcml(self):
+        note_path = os.path.join(TSV_PATH, "test_notes.tsv")
+        measure_path = os.path.join(TSV_PATH, "test_measures.tsv")
+        harmony_path = os.path.join(TSV_PATH, "test_harmonies.tsv")
+        score = load_dcml(note_path, measure_path, harmony_path)
+        self.assertEqual(len(score.parts), 1)
+
```
Comment (fosfrancesco) [inline tests/test_dcml_import.py:14]: maybe another assertion that counts the number of notes and compares with the number of lines in the tsv? Will they match?

**Review by fosfrancesco — CHANGES_REQUESTED** [review]

File: docs/source/Tutorial/notebook.ipynb
Comment (muthissar) [inline docs/source/Tutorial/notebook.ipynb:1]: Do not know if there was something substantial changed here?

File: partitura/utils/music.py (hunk 2 above)
Comment (manoskary) [reply partitura/utils/music.py:196]: It is not used anywhere else. I would like to keep it for now and delete it on the next release to make sure no errors with the new method arrise.

File: partitura/io/exportaudio.py [hunk 6]
```diff
@@ -1,8 +1,7 @@
 #!/usr/bin/python
 # -*- coding: utf-8 -*-
 """
-This module contains methods to synthesize Partitura object to wav using
-additive synthesis
+This module contains methods to synthesize Partitura object to wav.
```
Comment (muthissar) [inline partitura/io/exportaudio.py:4]: either:
-  This module contains methods to synthesize a Partitura object to wav.
-  This module contains methods to synthesize Partitura objects to wav.

File: partitura/io/exportkern.py [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+                return "q"
+        else:
+            if "type" not in element.symbolic_duration.keys():
+                warnings.warn(f"Element {element} has no symbolic duration type")
+                return "4"
+            return self.sym_dur_to_kern(element.symbolic_duration)
+
+    def pitch_to_kern(self, element: spt.GenericNote) -> str:
+        # To encode pitch correctly in kern we need to take into account the octave
+        # duplication of the step in kern can either move the note up or down an octave
```
Comment (muthissar) [inline partitura/io/exportkern.py:223]: # To encode pitch correctly in kern we need to take into account that the octave
        # duplication of the step in kern can either move the note up or down an octave

File: partitura/io/importdcml.py [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+
+try:
+    import pandas as pd
+except ImportError:
+    pd = None
+
+
+def read_note_tsv(note_tsv_path, metadata=None):
+    # data = np.genfromtxt(note_tsv_path, delimiter="\t", dtype=None, names=True, invalid_raise=False)
+    # unique_durations = np.unique(data["duration"])
```
Comment (muthissar) [inline partitura/io/importdcml.py:17]: Do these lines need to stay?

File: partitura/score.py [hunk 9]
```diff
@@ -13,14 +13,20 @@
...
 # import copy
-from partitura.utils.music import MUSICAL_BEATS, INTERVALCLASSES
+from partitura.utils.globals import (
+    MUSICAL_BEATS,
+    INTERVALCLASSES,
+    INTERVAL_TO_SEMITONES,
+)
 import warnings, sys
 import numpy as np
+import re
```
Comment (muthissar) [inline partitura/score.py:26]: This was already imported at line 16

File: tests/test_note_array.py [hunk 10]
```diff
@@ -175,9 +175,9 @@ def test_ensure_na_different_divs(self):
         # note_arrays = [p.note_array(include_divs_per_quarter= True) for p in parts]
         merged_note_array = ensure_notearray(parts)
         for note in merged_note_array[-4:]:
-            self.assertTrue(note["onset_div"] == 92)
-            self.assertTrue(note["duration_div"] == 4)
-            self.assertTrue(note["divs_pq"] == 4)
+            self.assertTrue(note["onset_div"] == 2208)
```
Comment (muthissar) [inline tests/test_note_array.py:178]: Not sure if it was necessary to change the test values?

**Review by muthissar — APPROVED** [review]
I am not able to say much, but I did not find any problems. A lot of updates to the core score file.

File: tests/test_note_array.py (hunk 10 above)
Comment (manoskary) [reply tests/test_note_array.py:178]: yes, new import changed order of notes.

File: partitura/score.py (hunk 9 above)
Comment (manoskary) [reply partitura/score.py:26]: thanks, good catch

**Review by fosfrancesco — APPROVED** [review]

**Review by fosfrancesco — APPROVED** [review]

**Pull request merged**